static/js/main.js

```diff
@@ -1107,26 +1107,40 @@ const addCopyButtons = (clipboard)=>{
         pre.appendChild(button);
     });
 };
-const addAnchorLinks = ()=>{
-    document.querySelectorAll(".content h1, .content h2, .content h3, .content h4").forEach((heading)=>{
-        let id = heading.innerText.toLowerCase().replace(/[`~!@#$%^&*()_|+=?;:'",.<>\{\}\[\]\\\/]/gi, "").replace(/ +/g, "-");
-        heading.setAttribute("id", id);
-        heading.classList.add("heading-anchor");
-        let anchor = document.createElement("a");
-        anchor.className = "anchor-link";
-        anchor.href = "#" + id;
+const addAnchorLinks = () => {
+    const elementsToProcess = document.querySelectorAll(".content h1, .content h2, .content h3, .content h4, .content tr");
+    const idMap = new Map(); // Map to store the counts of each ID
+    elementsToProcess.forEach(element => {
+        let id;
+        if (element.tagName.toLowerCase() === 'tr') {
+            const rowContent = element.cells[0].textContent.trim();
+            id = `${rowContent.replace(/[`~!@#$%^&*()_|+=?;:'",.<>\{\}\[\]\\\/]/gi, '').replace(/ +/g, '-')}`;
```

Comment:
if two cells have the same content, this will lead to an ID collision (which can happen if we have 2 tables). We should devise a way here to generate a consistent unique ID for the table.

Reply:
I think the code below ensures that the generated IDs for anchor links are unique, especially when there are cells in a table with the same content, right?
` let count = idMap.get(id) || 0;
        count++;
        idMap.set(id, count);
        if (count > 1) {
            id += ${count};
        }
`

Reply:
Yep I think that should be fine. My apologies I missed that.

Reply:
Are the IDs generated by a runtime counter going to be stable over time?

Reply:
I don't think so because If the content of the page changes, new elements are added, or existing elements are removed or modified, the counter may result in different IDs being generated.

Reply:
In that case I'm afraid this might need a rethink.  The purpose of this is to enable other content to link to the right parts of the table, and it's not generally convenient or even possible to update that other content if an anchor changes.  Sorry.